monai/apps/nuclick/transforms.py

```diff
@@ -0,0 +1,534 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import random
+from enum import Enum
+from typing import Any, Tuple, Union
+
+import numpy as np
+
+from monai.config import KeysCollection
+from monai.transforms import MapTransform, Randomizable, SpatialPad
+from monai.utils import optional_import
+
+measure, _ = optional_import("skimage.measure")
+morphology, _ = optional_import("skimage.morphology")
+
+
+class NuclickKeys(Enum):
+    """
+    Keys for nuclick transforms.
+    """
+
+    IMAGE = "image"
+    LABEL = "label"
+    OTHERS = "others"  # key of other labels from the binary mask which are not being used for training
+    FOREGROUND = "foreground"
+
+    CENTROID = "centroid"  # key where the centroid values are stored
+    MASK_VALUE = "mask_value"
+    LOCATION = "location"
+
+    NUC_POINTS = "nuc_points"
+    BOUNDING_BOXES = "bounding_boxes"
+    IMG_HEIGHT = "img_height"
+    IMG_WIDTH = "img_width"
+
+
+class FlattenLabeld(MapTransform):
+    """
+    FlattenLabeld creates labels per closed object contour (defined by a connectivity). For e.g if there are
+    12 small regions of 1's it will delineate them into 12 different label classes
+
+    Args:
+        connectivity: Max no. of orthogonal hops to consider a pixel/voxel as a neighbor. Refer skimage.measure.label
+        allow_missing_keys: don't raise exception if key is missing.
+    """
+
+    def __init__(self, keys: KeysCollection, connectivity: int = 1, allow_missing_keys: bool = False):
+        super().__init__(keys, allow_missing_keys)
+        self.connectivity = connectivity
+
+    def __call__(self, data):
+        d = dict(data)
+        for key in self.keys:
+            d[key] = measure.label(d[key], connectivity=self.connectivity).astype(np.uint8)
+        return d
+
+
+class ExtractPatchd(MapTransform):
+    """
+    Extracts a patch from the given image and label, however it is based on the centroid location.
+    The centroid location is a 2D coordinate (H, W). The extracted patch is extracted around the centroid,
+    if the centroid is towards the edge, the centroid will not be the center of the image as the patch will be
+    extracted from the edges onwards
+
+    Args:
+        keys: image, label
+        centroid_key: key where the centroid values are stored, defaults to ``"centroid"``
+        patch_size: size of the extracted patch
+        allow_missing_keys: don't raise exception if key is missing.
```

Comment:
Could you please help change the arg name to `pad_kwargs` and add the **doc-string & unit test** for it?

Thanks.